OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -407,4 +413,29 @@ public UnsavedWorkState canOpen() {
             return UnsavedWorkState.OPEN;
         }
     }
+
+    private void autoDiscoverLaTeXPath() {
+        try {
+            String command;
+
+            if (System.getProperty("os.name").contains("Windows")) {
+                command = "where " + ApplicationUtils.TEX_NAME;
+            } else {
+                command = "which " + ApplicationUtils.TEX_NAME;
+            }
+
+            Process p = Runtime.getRuntime().exec(command);

```

Comment:
Can we use Apache Commons Exec instead? (This is the library we use for invoking LaTeX commands)